erpnext/projects/doctype/timesheet/timesheet.py

```diff
@@ -28,6 +28,7 @@ def validate(self):
 		self.update_cost()
 		self.calculate_total_amounts()
 		self.calculate_percentage_billed()
+		self.set_dates()
```

Comment:
the idea behind this is to keep a consitancy seems wierd to not have on save only submit and i wanted to use for a customisation but i could easily add another way